lightning/src/ln/functional_tests.rs

```diff
@@ -4182,7 +4184,7 @@ fn test_static_spendable_outputs_justice_tx_revoked_htlc_timeout_tx() {
 	check_closed_broadcast!(nodes[1], false);
 
 	let node_txn = nodes[1].tx_broadcaster.txn_broadcasted.lock().unwrap();
-	assert_eq!(node_txn.len(), 4 ); // ChannelMonitor: justice tx on revoked commitment, justice tx on revoked HTLC-timeout, adjusted justice tx, ChannelManager: local commitment tx
+	assert_eq!(node_txn.len(), 5); // ChannelMonitor: justice tx on revoked commitment, justice tx on revoked HTLC-timeout, adjusted justice tx, bumped justice tx, ChannelManager: local commitment tx
```

Comment:
Please try to avoid checking for transaction presense without checking what that transaction is. Either check_spends!() it or check the witness length, at least.

Reply:
Ooooh thanks for acute comment, in fact fixing bug by tracking revoked htlc output was adding another bug by duplicating regeneration of adjusted claim transaction (aka when an aggregated tx has one of its outpoints claimed we MUST regenerate a new valid claim tx). See new commit for more context.